src/encoded/types/file.py

```diff
@@ -378,6 +382,59 @@ def create(cls, registry, uuid, properties, sheets=None):
             sheets['external'] = external_creds(bucket, key, name, profile_name)
         return super(File, cls).create(registry, uuid, properties, sheets)
 
+    def _get_external_sheet(self):
+        external = self.propsheets.get('external', {})
+        if external.get('service') == 's3':
+            return external
+        else:
+            raise HTTPNotFound()
+
+    def set_public_s3(self):
+        external = self._get_external_sheet()
+        boto3.resource('s3').ObjectAcl(
+            external['bucket'],
+            external['key']
+        ).put(ACL='public-read')
+
+    def set_private_s3(self):
+        external = self._get_external_sheet()
+        boto3.resource('s3').ObjectAcl(
+            external['bucket'],
+            external['key']
+        ).put(ACL='private')
+
+    def set_status(self, new_status, request, parent=True):
+        properties = self.upgrade_properties()
+        status = properties.get('status')
+        # Is valid transition?
+        if status not in STATUS_TRANSITION_TABLE[new_status]:
+            # Raise failure if this is primary object.
+            if parent:
```

Comment:
So if parent=False there is no error even if the transition isn't allowed?  This is kind of confusing.  I suspect this is just an un-implemented feature.

Reply:
or it raises and error otherwise silently does nothing?

Reply:
Unimplemented. The idea is that if you are trying to change status of primary object it should fail if it's not a valid a transition. However if we add code to consider child objects after considering the primary object (e.g. releasing files in an experiment after releasing the experiment) then it shouldn't necessarily fail if one of the child objects isn't a valid transition.